CONTRIBUTING.md

```diff
@@ -7,23 +7,211 @@ repository before making a change.
+    // "coverage-gutters.showLineCoverage": true,
+  },
+  "launch": {
+    "configurations": [
+      {
+        // Sometimes I need to run mocha myself and not via the sidebar
+        // Here I can add custom args or env variables
+        "name": "run mocha",
+        "type": "pwa-node",
+        "request": "launch",
+        "program": "node_modules/.bin/mocha",
+        "args": ["test/unit", "test/functional"]
+      }
+    ],
+    "compounds": []
+  },
+  "tasks": {
+    "version": "2.0.0",
+    "tasks": [
+      {
+        // Here is an optional watcher task (`npm test` will also type check you changes):
+        // Since this is the default build task it can be started with cmd+shift+b
+        // There will be a wrench and screw icon
+        // on the bottom bar where you can quick check build issues
+        "label": "watch TS",
+        "command": "npx",
+        "type": "shell",
+        "args": ["tsc", "-w"],
+        "problemMatcher": "$tsc-watch",
+        "isBackground": true,
+        "group": {
+          "kind": "build",
+          "isDefault": true
+        }
+      }
+    ]
+  },
+  "extensions": {
+    "recommendations": [
+      "dbaeumer.vscode-eslint",
+      "hbenl.vscode-test-explorer",
+      "hbenl.vscode-mocha-test-adapter",
+      "ryanluker.vscode-coverage-gutters",
+      "github.vscode-pull-request-github",
+      "mongodb.mongodb-vscode"
+    ],
+    "unwantedRecommendations": ["esbenp.prettier-vscode"]
+  }
+}
+</pre>
+</details>
+
+If you use this file you will get our recommended extensions suggested to you. If not, we recommend picking up `dbaeumer.vscode-eslint` at least to make sure any additional code is following style recommendations.
+
+```jsonc
+"editor.codeActionsOnSave": {
+  "source.fixAll.eslint": true
+},
+"[javascript]": {
+  "editor.defaultFormatter": "dbaeumer.vscode-eslint"
+},
+"[typescript]": {
+  "editor.defaultFormatter": "dbaeumer.vscode-eslint",
+}
+```
+
+### Running the tests
+
+Running the tests:
+
+- You can use the script: `test/tools/cluster_setup.sh server`
+- If you are running against more than a standalone make sure your ulimit settings are in accordance with mongo's recommendations
+  - Changing the settings on the latest versions of [macos can be tricky read here][macos-ulimt] (unless you know you need it you shouldn't have to do the complicated maxproc steps)
+- Prefix the cluster_setup.sh script with `env MONGODB_VERSION=X.Y` to test against a specific version of the server
+- `env MONGODB_URI=mongodb://localhost:27017 npm test`
+- When testing different topologies you may need to remove the existing data folder created.
+- To run a single test, use `npm run test -- -g 'test name'`
+  - If it's easier you can also isolate tests by adding .only. Example: `it.only(‘cool test’, {})`
+- To test only the unified topology, use `env MONGODB_UNIFIED_TOPOLOGY=1`
+
+### Commit messages
 
 Please follow the [Angular commit style][angular-commit-style].
```

Comment:
I recently (yesterfday) found out this format has a name "Conventional Commits" https://www.conventionalcommits.org/en/v1.0.0/ perhaps point to that? There's also some pretty cool tooling around this https://github.com/conventional-changelog/commitlint.

Here's the example config / dependencies if we ever want reference:

```json
{
  "name": "@reggi/modules",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "commitmsg": "commitlint -e $GIT_PARAMS",
    "bootstrap-ci": "lerna bootstrap --ci",
    "test-changed": "lerna run test --since main",
    "build-changed": "lerna run build --since main",
    "publish": "lerna publish -y"
  },
  "devDependencies": {
    "@commitlint/cli": "^7.6.1",
    "@commitlint/config-conventional": "^7.6.0",
    "husky": "^0.14.3",
    "hygen": "^6.0.4",
    "lerna": "^3.22.1"
  },
  "commitlint": {
    "extends": [
      "@commitlint/config-conventional"
    ]
  }
}
```

Reply:
Talked a bit about this in the channel and it on the schedule for our meeting on Monday.
Summary seemed to be: that we are interested in using this but it won't help verify commit made through github UI which is our primary workflow.